hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequestWithFSO.java

```diff
@@ -122,7 +121,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       validateBucketAndVolume(omMetadataManager, volumeName, bucketName);
 
       String fileName = OzoneFSUtils.getFileName(keyName);
-      omBucketInfo = omMetadataManager.getBucketTable().get(bucketKey);
+      omBucketInfo = getBucketInfo(omMetadataManager, volumeName, bucketName);
```

Comment:
In OMKeyCommitRequest and other request(like OMKeyCreateRequest) we use getBucketInfo to get omBucketInfo.   I suspect the problem is that the objects we get from getBucketTable are not the same as those we get from getBucketInfo.

Reply:
The cache is a `Map`. By getting omBucketInfo from cache, when we modify it later, we are also modifying the object in cache.